sql/stats: TestAtMostOneRunningCreateStats failed

sql/stats.TestAtMostOneRunningCreateStats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21038003?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21038003?buildTab=artifacts#/) on master @ [e2a6bb07b7cea7b02ede8a6087d4d2515d789681](https://github.com/cockroachdb/cockroach/commits/e2a6bb07b7cea7b02ede8a6087d4d2515d789681):


```
=== RUN   TestAtMostOneRunningCreateStats
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/105639e676f402ac2c172927559216df/logTestAtMostOneRunningCreateStats377714822
    test_log_scope.go:82: use -show-logs to present logs inline
    create_stats_job_test.go:958: auto stats job should have failed, but it didn't (beforeCount: 0, afterCount: 1)
    panic.go:615: -- test log scope end --
--- FAIL: TestAtMostOneRunningCreateStats (2.18s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAtMostOneRunningCreateStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59203

sql/stats.TestAtMostOneRunningCreateStats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21038455?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21038455?buildTab=artifacts#/) on master @ [704c1b2e6fb0499a2f7aa64356ee3d6c169bbd35](https://github.com/cockroachdb/cockroach/commits/704c1b2e6fb0499a2f7aa64356ee3d6c169bbd35):


```
=== RUN   TestAtMostOneRunningCreateStats
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/105639e676f402ac2c172927559216df/logTestAtMostOneRunningCreateStats1890250145
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:679: DRPC is enabled (via override by TestingGlobalDRPCOption)
    create_stats_job_test.go:958: auto stats job should have failed, but it didn't (beforeCount: 0, afterCount: 1)
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/105639e676f402ac2c172927559216df/logTestAtMostOneRunningCreateStats1890250145
--- FAIL: TestAtMostOneRunningCreateStats (1.54s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAtMostOneRunningCreateStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/stats.TestAtMostOneRunningCreateStats [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21038957?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21038957?buildTab=artifacts#/) on master @ [c92b2cff04d81f01ff7cd3f252e78895f6474f3d](https://github.com/cockroachdb/cockroach/commits/c92b2cff04d81f01ff7cd3f252e78895f6474f3d):


```
=== RUN   TestAtMostOneRunningCreateStats
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/105639e676f402ac2c172927559216df/logTestAtMostOneRunningCreateStats3045438311
    test_log_scope.go:82: use -show-logs to present logs inline
    create_stats_job_test.go:958: auto stats job should have failed, but it didn't (beforeCount: 0, afterCount: 1)
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/105639e676f402ac2c172927559216df/logTestAtMostOneRunningCreateStats3045438311
--- FAIL: TestAtMostOneRunningCreateStats (0.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAtMostOneRunningCreateStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/stats/create_stats_job_test.go
+++ b/pkg/sql/stats/create_stats_job_test.go
@@ -221,12 +221,10 @@ func TestCreateStatisticsCanBeCancelled(t *testing.T) {
 // don't run when an auto full stats job is running. It also tests that manual
 // stat jobs (full or partial) are always allowed to run.
 func TestAtMostOneRunningCreateStats(t *testing.T) {
-	skip.WithIssue(t, 161973)
 	testAtMostOneRunningCreateStatsImpl(t, false /* shouldError */)
 }
 
 func TestAtMostOneRunningCreateStatsWithErrorOnConcurrentCreateStats(t *testing.T) {
-	skip.WithIssue(t, 161974)
 	testAtMostOneRunningCreateStatsImpl(t, true /* shouldError */)
 }
 
@@ -262,6 +260,8 @@ func testAtMostOneRunningCreateStatsImpl(t *testing.T, shouldError bool) {
 	// Disable automatic cleanup of completed jobs since we might block on a job
 	// until it succeeds.
 	sqlDB.Exec(t, `SET CLUSTER SETTING sql.stats.automatic_stats_job_auto_cleanup.enabled = false`)
+	// Lower the concurrency limit to allow at most one auto full stats job.
+	sqlDB.Exec(t, `SET CLUSTER SETTING sql.stats.automatic_full_concurrency_limit = 1`)
 	sqlDB.Exec(t, fmt.Sprintf("SET CLUSTER SETTING sql.stats.error_on_concurrent_create_stats.enabled = %t", shouldError))
 	sqlDB.Exec(t, `CREATE DATABASE d`)
 	sqlDB.Exec(t, `CREATE TABLE d.t (x INT PRIMARY KEY)`)

